### DISCUSSION TITLE
Arrow

### DISCUSSION THREAD

**Discussion by parapenT1sta** [jakobheyman/xdripJH#5] [General]
Hi,
I started using this xDrip version for L2 but there something making me annoyed that is the Graphical Trend Arrow. When I have delta of +6mg/dl  is showing me the equivalent of double arrow up. Is anyway to fix it or is it the normal behaviour?
I understand that +6 for 5 minutes will be equivalent to +30, but should not be taken in count the delta on that moment and not for the 5 minutes? For example +30 from last reading (last 1 minute) be equivalent to double arrow?

**Comment by jakobheyman** [comment]
Thanks for your comment!
For me, the graphical trend arrows are generally reasonable and I believe (although I might be wrong) that I have not modified the delta or arrows. I think the delta is calculated based on the last two blood glucose values but recalculated to give the hypothetical change over exactly five minutes. Hence, I think that the graphical trend arrows should be the same as they would in the main xDrip app with a similar blood glucose change. However, as it has worked good for me I have never really looked into delta and arrow issues.

**Comment by parapenT1sta** [comment]
Thank you for replying.
I thought xdrip delta was based delta from the previous reading the the last reading. Can you check it?

**Comment by jakobheyman** [comment]
Yes - I think the delta is based on the difference between the previous and the last reading but it is also related to the time difference between the two [(see here)](https://github.com/jakobheyman/xdripJH/blob/bf42df565c7a3882cce16ff41f709a808a1912a6/app/src/main/java/com/eveningoutpost/dexdrip/models/BgReading.java#L318C1-L318C110). I think there are a couple of other reasons why the arrows may be annoying in xdripJH:
1. If using Libre-2 with OOP2 in xdripJH there is no smoothing of the bg values - in the main app xDrip I think there is.
2. With 1 minute data, a jump in the bg values that can seem relatively small can still result in a rather big jump in the delta as the 1 minute difference is extrapolated to 5 minutes. That does not happen as easily if using 5 minute intervals.
